File: JAVA/Data Structures/Linear/Queue/src/Queue.java
public class Queue {
    private int[] myQueue = new int[5];
    private int firstIn;
    private int firstOut;

    public void enqueue(int num) {
        myQueue[firstIn] = num;
        firstIn++;
    }

    public int dequeue() {
        int num = myQueue[firstOut];
        firstOut++;
        return num;
    }

    public int peek() {
        return myQueue[firstOut];
    }

    public void show() {
        int i = firstOut;
        while (i != firstIn) {
            System.out.print(myQueue[i] + " ");
            i++;
        }
        System.out.println();
    }
}

File: PYTHON/Data Structures/Linear/Queue/Queue.py
myQueue = [0] * 5

firstIn = 0
firstOut = 0

def enqueue(num):
    global firstIn

    myQueue[firstIn] = num
    firstIn += 1

def dequeue():
    global firstOut

    num = myQueue[firstOut]
    firstOut += 1

    return num

def peek():
    num = myQueue[firstOut]
    return num

def show():
    i = firstOut
    while(i != firstIn):
        print(myQueue[i])
        i += 1